src/lighteval/models/tgi_model.py

```diff
@@ -20,19 +20,12 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-import asyncio
-from typing import Coroutine, Optional
-
 import requests
-from huggingface_hub import TextGenerationInputGrammarType, TextGenerationOutput
+from huggingface_hub import AsyncInferenceClient, InferenceClient
 from transformers import AutoTokenizer
 
-from lighteval.models.endpoint_model import InferenceEndpointModel, ModelInfo
-from lighteval.utils.imports import NO_TGI_ERROR_MSG, is_tgi_available
-
-
-if is_tgi_available():
```

Comment:
TGI recommends using `huggnigface_hub` instead of `text-generation`.
https://github.com/huggingface/text-generation-inference/tree/main/clients/python